### PR TITLE
Added docker options

### PR DESCRIPTION
**Pull request by mhald**

### CODE REVIEW

File: docs/packaging-an-application/components-and-containers.md [hunk 1]
```diff
@@ -303,3 +303,37 @@ The startup section of a container allows you to specify the CMD value that will
...
 
 As with all inputs to containers, you have full access to the Replicated template library when creating a CMD value.
+
+## Docker Options
+You may also limit the resources used by your containers with the memory, cpushares and network modes and further secure your containers with security options
+
+### Memory Limit
+The amount of memory for your container.  The format is number|unit where unit may be one of b, k, m or g.  By default there is no memory limit and your container can use as much as needed.
+```yml
+  memory_limit: 500M
```
Comment (divolgin) [inline docs/packaging-an-application/components-and-containers.md]: Is it case insensitive?  The text above uses lower case.

File: docs/packaging-an-application/components-and-containers.md [hunk 2]
```diff
@@ -303,3 +303,37 @@ The startup section of a container allows you to specify the CMD value that will
...
+```
+
+### Memory Swap Limit
+The amount of swap memory allowed for your container.  The format is number|unit where unit may be one of b, k, m or g.  By default there is no swap limit.
+```yml
+  memory_swap_limit: 1G
```
Comment (divolgin) [inline docs/packaging-an-application/components-and-containers.md]: Same thing here

File: docs/packaging-an-application/components-and-containers.md [hunk 3]
```diff
@@ -303,3 +303,37 @@ The startup section of a container allows you to specify the CMD value that will
...
+Network mode supports bridge, host, container or none.
+```yml
+  network_mode: host
+```
+
+### Security Options
+With security options you can use Docker security with existing well know systems such as apparmor.  When specifying your security options you can use template functions and any blank security option is allowed and will be filtered out by Replicated.
+```yml
+security_options:
+  - apparmor=unconfined
```
Comment (divolgin) [inline docs/packaging-an-application/components-and-containers.md:333]: Maybe add another option that shows how to use a template?

**Comment by marccampbell** [comment]
@mhald i think we should link to the Docker docs on each of these items for more info.

**Comment by marccampbell** [comment]
@mhald let's also add the other fields we support that are really just passed straight through to docker:
- privileged
- network_mode
- allocate_tty
- security_cap_add
- hostname
- extra_hosts

**Pull request merged**